### PR TITLE
Tutorial on Cell type identification from scRNA-seq data using ACTINN and Deepchem

### PR DESCRIPTION
**Pull request by Harindhar10**
## Description

Fix #(issue)

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [ ] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [ ] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [ ] Run `mypy -p deepchem` and check no errors
  - [ ] Run `flake8 <modified file> --count` and check no errors
  - [ ] Run `python -m doctest <modified file>` and check no errors
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: examples/tutorials/Cell_type_identification_using_scRNAseq_data.ipynb [hunk 1]
```diff
@@ -0,0 +1,608 @@
+{
```
Comment (sudo-rsingh) [inline examples/tutorials/Cell_type_identification_using_scRNAseq_data.ipynb:1]: <p>can you use a for loop for this</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4496/discussion/#comment-2222134502'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (sudo-rsingh) [inline examples/tutorials/Cell_type_identification_using_scRNAseq_data.ipynb:1]: <p>Pls add citation for Tabula Muris</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4496/discussion/#comment-2222134505'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (sudo-rsingh) [inline examples/tutorials/Cell_type_identification_using_scRNAseq_data.ipynb:1]: <p>Pls add more theory about the ADCNet model</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4496/discussion/#comment-2222134510'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='30'/>

Comment (sudo-rsingh) [inline examples/tutorials/Cell_type_identification_using_scRNAseq_data.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #23.            self.classifier_sequential = nn.Sequential(
</pre><p>can you pls use deepchem's MultiLayerPerceptrons here</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4496/discussion/#comment-2222134515'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='30'/>

Comment (sudo-rsingh) [inline examples/tutorials/Cell_type_identification_using_scRNAseq_data.ipynb:1]: <p>Can you please run these sections, so we can have some idea how its running</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4496/discussion/#comment-2222134519'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='36'/>

Comment (sudo-rsingh) [inline examples/tutorials/Cell_type_identification_using_scRNAseq_data.ipynb:1]: <p>Pls add all the references, and notebook author details, how to cite and deepchem footer</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4496/discussion/#comment-2222134520'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='37'/>

Comment (Harindhar10) [reply examples/tutorials/Cell_type_identification_using_scRNAseq_data.ipynb:1]: <p>fixed</p>

Comment (Harindhar10) [reply examples/tutorials/Cell_type_identification_using_scRNAseq_data.ipynb:1]: <p>added</p>

Comment (Harindhar10) [reply examples/tutorials/Cell_type_identification_using_scRNAseq_data.ipynb:1]: <p>Deepchem's MLP applies activation function(ReLU in this case) to the output layer, and I couldn't find a way to change that. nn.CrossEntropyLoss expects raw logits and applies Softmax. When ReLU is applied before Softmax, wouldn't it make negative values zero and affect the relative difference between classes?</p>

Comment (Harindhar10) [reply examples/tutorials/Cell_type_identification_using_scRNAseq_data.ipynb:1]: done

Comment (Harindhar10) [reply examples/tutorials/Cell_type_identification_using_scRNAseq_data.ipynb:1]: done

Comment (Harindhar10) [reply examples/tutorials/Cell_type_identification_using_scRNAseq_data.ipynb:1]: <p>done</p>

Comment (Harindhar10) [reply examples/tutorials/Cell_type_identification_using_scRNAseq_data.ipynb:1]: <p>done</p>

File: examples/tutorials/Cell_type_identification_using_scRNAseq_data.ipynb [hunk 2]
```diff
@@ -0,0 +1,1698 @@
...
```
Comment (Sonalilipsa17375) [inline examples/tutorials/Cell_type_identification_using_scRNAseq_data.ipynb:1]: <p>I think a bit of background context in the start could be a good addition.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4496/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (Sonalilipsa17375) [inline examples/tutorials/Cell_type_identification_using_scRNAseq_data.ipynb:1]: <p>I think this needs a bit of formatting. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4496/discussion/#comment-2235066617'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='60'/>

Comment (Sonalilipsa17375) [inline examples/tutorials/Cell_type_identification_using_scRNAseq_data.ipynb:1]: <p><span class="mention" data-value="Harindhar10" data-denotation-char="@">﻿<span contenteditable="false">@Harindhar10</span>﻿</span> the colab link here is redirecting to another tutorial i.e. ADvanced model training tut.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4496/discussion/#comment-2235072588'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='2'/>

Comment (Harindhar10) [reply examples/tutorials/Cell_type_identification_using_scRNAseq_data.ipynb:1]: <p>fixed</p>

Comment (Harindhar10) [reply examples/tutorials/Cell_type_identification_using_scRNAseq_data.ipynb:1]: fixed

File: examples/tutorials/Cell_type_identification_using_scRNAseq_data.ipynb [hunk 3]
```diff
@@ -0,0 +1,1629 @@
...
```
Comment (sudo-rsingh) [inline examples/tutorials/Cell_type_identification_using_scRNAseq_data.ipynb:1]: <p>A more detailed background should be here</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4496/discussion/#comment-2243162565'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (sudo-rsingh) [inline examples/tutorials/Cell_type_identification_using_scRNAseq_data.ipynb:1]: <p>Pls not use links, use traditional citations</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4496/discussion/#comment-2243162571'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (sudo-rsingh) [inline examples/tutorials/Cell_type_identification_using_scRNAseq_data.ipynb:1]: <p>Pls add more details about the model implementation</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4496/discussion/#comment-2243162579'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='49'/>

Comment (sudo-rsingh) [inline examples/tutorials/Cell_type_identification_using_scRNAseq_data.ipynb:1]: <p>Full author names should be here and </p><p> </p><p>.ipynb is typed twice</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4496/discussion/#comment-2243162586'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='58'/>

Comment (Harindhar10) [reply examples/tutorials/Cell_type_identification_using_scRNAseq_data.ipynb:1]: <p>added</p>

Comment (Harindhar10) [reply examples/tutorials/Cell_type_identification_using_scRNAseq_data.ipynb:1]: <p>fixed</p>

File: examples/tutorials/Cell_type_identification_using_scRNAseq_data.ipynb [hunk 4]
```diff
@@ -0,0 +1,1623 @@
...
```
Comment (rbharath) [inline examples/tutorials/Cell_type_identification_using_scRNAseq_data.ipynb:1]: <p>References need to be cited inline in the tutorial above</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4496/discussion/#comment-2243361001'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='55'/>

File: examples/tutorials/Cell_type_identification_using_scRNAseq_data.ipynb (hunk 3 above)
Comment (Harindhar10) [reply examples/tutorials/Cell_type_identification_using_scRNAseq_data.ipynb:1]: added

Comment (Harindhar10) [reply examples/tutorials/Cell_type_identification_using_scRNAseq_data.ipynb:1]: fixed

File: examples/tutorials/Cell_type_identification_using_scRNAseq_data.ipynb (hunk 4 above)
Comment (Harindhar10) [reply examples/tutorials/Cell_type_identification_using_scRNAseq_data.ipynb:1]: cited
